api/configuration/v1alpha1/kong_certificate.go

```diff
@@ -0,0 +1,82 @@
+package v1alpha1
+
+import (
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KongCertificate is the schema for Certificate API which defines a Kong Certificate.
+//
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Programmed",description="The Resource is Programmed on Konnect",type=string,JSONPath=`.status.conditions[?(@.type=='Programmed')].status`
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.spec.controlPlaneRef) || has(self.spec.controlPlaneRef)", message="controlPlaneRef is required once set"
+// +kubebuilder:validation:XValidation:rule="(!self.status.conditions.exists(c, c.type == 'Programmed' && c.status == 'True')) ? true : oldSelf.spec.controlPlaneRef == self.spec.controlPlaneRef", message="spec.controlPlaneRef is immutable when an entity is already Programmed"
+type KongCertificate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec KongCertificateSpec `json:"spec"`
+
+	// +kubebuilder:default={conditions: {{type: "Programmed", status: "Unknown", reason:"Pending", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}}
+	Status KongCertificateStatus `json:"status,omitempty"`
+}
+
+// KongCertificateSpec contains the specification for the KongCertificate.
+type KongCertificateSpec struct {
+	// ControlPlaneRef references the Konnect Control Plane that this KongCertificate should be created in.
+	ControlPlaneRef        *ControlPlaneRef `json:"controlPlaneRef,omitempty"`
+	KongCertificateAPISpec `json:",inline"`
+}
+
+// KongCertificateAPISpec contains the API specification for the KongCertificate.
+type KongCertificateAPISpec struct {
```

Comment:
Aren't we missing the `ID` field in here as per https://kong-platform-api.netlify.app/konnect/control-planes-config/v2/openapi.yaml/#tag/Certificates/operation/create-certificate ?

Reply:
Hm I've just noticed now that we don't set that field when creating resources on Konnect using the SDK , e.g. 

- https://github.com/Kong/gateway-operator/blob/b449bc72d51a14c26bab4fe9bbbffefb13b93c2c/controller/konnect/ops/ops_kongconsumer.go#L337-L341
- https://github.com/Kong/gateway-operator/blob/b449bc72d51a14c26bab4fe9bbbffefb13b93c2c/controller/konnect/ops/ops_kongroute.go#L125-L146

even though the fields are in place:
- https://github.com/Kong/sdk-konnect-go/blob/077efc64e23b051ebff780fd2c05b5113463f74f/models/components/consumerinput.go#L8
- https://github.com/Kong/sdk-konnect-go/blob/077efc64e23b051ebff780fd2c05b5113463f74f/models/components/route.go#L385

Let's keep it that way for certificate as well.